webapp/src/main/java/com/box/l10n/mojito/service/tm/TMService.java

```diff
@@ -783,6 +798,27 @@ public boolean isUpdateNeededForTmTextUnitVariant(
         && Objects.equals(currentComment, newComment));
   }
 
+  /**
+   * Indicates if only the {@link TMTextUnitVariant.Status} and includedInLocalizedFile boolean has
+   * changed to require an update for a {@link TMTextUnitVariant} by looking at new/old content,
+   * status, comments, etc
+   *
+   * <p>Used to avoid causing Auditing table update failures as a variant status change does not
+   * require the creation of a new variant.
+   *
+   * @param currentContentMd5
+   * @param currentComment
+   * @param newContentMd5
+   * @param newComment
+   * @return
+   */
+  public boolean isContentOrCommentUpdated(
+      String currentContentMd5, String currentComment, String newContentMd5, String newComment) {
+
+    return !(currentContentMd5.equals(newContentMd5)
+        || !Objects.equals(currentComment, newComment));
```

Comment:
Why are we using object equality for these two strings? This seems a bit unusual

Reply:
It's to ensure we don't run String.equals on a null string, this way ensures if the currentComment string is null that an NPE won't occur and that we identify if it has been changed